src/repos/ArticleRepo.ts

```diff
@@ -2,29 +2,31 @@ import { ObjectId } from 'mongodb';
 import { Article, ArticleModel } from '../entities/Article';
 import Constants from '../common/constants';
 import getRecentArticles from '../db/rss-parser';
+import { PublicationModel } from '../entities/Publication';
 
 const getArticleByID = async (id: string): Promise<Article> => {
   return ArticleModel.findById(new ObjectId(id));
 };
 
 const getArticlesByIDs = async (ids: string[]): Promise<Article[]> => {
-  return Promise.
-    all(
-      ids.map((id) => ArticleModel.findById(new ObjectId(id)))
-    ).then((articles) => {
-      return articles;
-    });
+  return Promise.all(ids.map((id) => ArticleModel.findById(new ObjectId(id)))).then((articles) => {
+    return articles[0] != null ? articles : [];
```

Comment:
why are you checking `articles[0] != null` instead of if the array is empty or not?https://stackoverflow.com/questions/24403732/how-to-check-if-array-is-empty-or-does-not-exist

Reply:
will change